docs/sources/mimir/manage/tools/mimirtool.md

```diff
@@ -48,6 +48,10 @@ Mimirtool is a command-line tool that operators and tenants can use to execute a
 
   For more information about the `backfill` command, refer to [Backfill]({{< relref "#backfill" >}})
 
+- The `template` command enables you to render your alertmanager template.
```

Comment:
Instead of adding a new top-level command `template`, please make `render` be a subcommand of `alertmanager`: `mimirtool alertmanager render <...>`

Reply:
I got your point but it might be confusing since it's doesn't do any interaction with the alertmanager

Reply:
Moved 
```bash
alertmanager render --template-glob=TEMPLATE-GLOB --template-text=TEMPLATE-TEXT [<flags>]
    Render a given definition in a template file to standard output.

nabil@laptop:~/mimir# go run cmd/mimirtool/main.go alertmanager --help
usage: mimirtool alertmanager [<flags>] <command> [<args> ...]

View and edit Alertmanager configurations that are stored in Grafana Mimir.


Flags:
  --[no-]help                 Show context-sensitive help (also try --help-long and --help-man).
  --log.level="info"          set level of the logger
  --push-gateway.endpoint=PUSH-GATEWAY.ENDPOINT
                              url for the push-gateway to register metrics
  --push-gateway.job=PUSH-GATEWAY.JOB
                              job name to register metrics
  --push-gateway.interval=1m  interval to forward metrics to the push gateway
  --user=""                   Basic auth API user to use when contacting Grafana Mimir; alternatively, set MIMIR_API_USER. If empty, MIMIR_TENANT_ID is used instead. ($MIMIR_API_USER)
  --key=""                    Basic auth API key to use when contacting Grafana Mimir; alternatively, set MIMIR_API_KEY. ($MIMIR_API_KEY)
  --tls-ca-path=""            TLS CA certificate to verify Grafana Mimir API as part of mTLS; alternatively, set MIMIR_TLS_CA_PATH. ($MIMIR_TLS_CA_PATH)
  --tls-cert-path=""          TLS client certificate to authenticate with the Grafana Mimir API as part of mTLS; alternatively, set MIMIR_TLS_CERT_PATH. ($MIMIR_TLS_CERT_PATH)
  --tls-key-path=""           TLS client certificate private key to authenticate with the Grafana Mimir API as part of mTLS; alternatively, set MIMIR_TLS_KEY_PATH. ($MIMIR_TLS_KEY_PATH)
  --[no-]tls-insecure-skip-verify
                              Skip TLS certificate verification; alternatively, set MIMIR_TLS_INSECURE_SKIP_VERIFY. ($MIMIR_TLS_INSECURE_SKIP_VERIFY)
  --auth-token=""             Authentication token bearer authentication; alternatively, set MIMIR_AUTH_TOKEN. ($MIMIR_AUTH_TOKEN)
  --[no-]utf8-strict-mode     Enable UTF-8 strict mode. Allows UTF-8 characters in the matchers for routes and inhibition rules, in silences, and in the labels for alerts.

Subcommands:
alertmanager get --address=ADDRESS --id=ID [<flags>]
    Get the Alertmanager configuration that is currently in the Grafana Mimir Alertmanager.

alertmanager delete --address=ADDRESS --id=ID
    Delete the Alertmanager configuration that is currently in the Grafana Mimir Alertmanager.

alertmanager load --address=ADDRESS --id=ID <config> [<template-files>...]
    Load Alertmanager tenant configuration and template files into Grafana Mimir.

alertmanager verify <config> [<template-files>...]
    Verify Alertmanager tenant configuration and template files.

alertmanager render --template-glob=TEMPLATE-GLOB --template-text=TEMPLATE-TEXT [<flags>]
    Render a given definition in a template file to standard output.


nabil@laptop:~/mimir# go run cmd/mimirtool/main.go alertmanager  render --help
usage: mimirtool alertmanager render --template-glob=TEMPLATE-GLOB --template-text=TEMPLATE-TEXT [<flags>]

Render a given definition in a template file to standard output.


Flags:
  --[no-]help                    Show context-sensitive help (also try --help-long and --help-man).
  --log.level="info"             set level of the logger
  --push-gateway.endpoint=PUSH-GATEWAY.ENDPOINT
                                 url for the push-gateway to register metrics
  --push-gateway.job=PUSH-GATEWAY.JOB
                                 job name to register metrics
  --push-gateway.interval=1m     interval to forward metrics to the push gateway
  --user=""                      Basic auth API user to use when contacting Grafana Mimir; alternatively, set MIMIR_API_USER. If empty, MIMIR_TENANT_ID is used instead. ($MIMIR_API_USER)
  --key=""                       Basic auth API key to use when contacting Grafana Mimir; alternatively, set MIMIR_API_KEY. ($MIMIR_API_KEY)
  --tls-ca-path=""               TLS CA certificate to verify Grafana Mimir API as part of mTLS; alternatively, set MIMIR_TLS_CA_PATH. ($MIMIR_TLS_CA_PATH)
  --tls-cert-path=""             TLS client certificate to authenticate with the Grafana Mimir API as part of mTLS; alternatively, set MIMIR_TLS_CERT_PATH. ($MIMIR_TLS_CERT_PATH)
  --tls-key-path=""              TLS client certificate private key to authenticate with the Grafana Mimir API as part of mTLS; alternatively, set MIMIR_TLS_KEY_PATH. ($MIMIR_TLS_KEY_PATH)
  --[no-]tls-insecure-skip-verify
                                 Skip TLS certificate verification; alternatively, set MIMIR_TLS_INSECURE_SKIP_VERIFY. ($MIMIR_TLS_INSECURE_SKIP_VERIFY)
  --auth-token=""                Authentication token bearer authentication; alternatively, set MIMIR_AUTH_TOKEN. ($MIMIR_AUTH_TOKEN)
  --[no-]utf8-strict-mode        Enable UTF-8 strict mode. Allows UTF-8 characters in the matchers for routes and inhibition rules, in silences, and in the labels for alerts.
  --template-glob=TEMPLATE-GLOB ...
                                 Glob of paths that will be expanded and used for rendering.
  --template-text=TEMPLATE-TEXT  The template that will be rendered.
  --template-type=TEMPLATE-TYPE  The type of the template. Can be either text (default) or html.
  --template-data=TEMPLATE-DATA  Full path to a file which contains the data of the alert(-s) with which the --template-text will be rendered. Must be in JSON. File must be formatted
                                 according to the following layout: https://pkg.go.dev/github.com/prometheus/alertmanager/template#Data. If none has been specified then a predefined,
                                 simple alert will be used for rendering.
  --id=""                        Basic auth username to use when contacting Prometheus or Grafana Mimir, also set as tenant ID; alternatively, set MIMIR_TENANT_ID. ($MIMIR_TENANT_ID)

```